R/detect_column.r

```diff
@@ -0,0 +1,354 @@
+  } else if (domain_spec=="SM"){
+    data("SM")
+    domain_spec <- force(SM)
+  } else if (domain_spec=="SV"){
+    data("SV")
+    domain_spec <- force(SV)
+  }
+  data("terminology")
+  terminology <- force(terminology)
+  term <- unlist(regmatches(domain_spec$'Controlled Terms, Codelist or Format', 
+                            gregexpr("(?=\\().*?(?<=\\))", 
+                                     domain_spec$'Controlled Terms, Codelist or Format', perl=T)))
+  term2 <- gsub("[()]","",term)
+  dat_term <- terminology[terminology$'CDISC Submission Value' %in% term2,]
+  dat_term2 <- terminology[terminology$'Codelist Code' %in% dat_term$Code,]
+  dat_term$`Codelist Code` <- NULL
+  names(dat_term)[names(dat_term) =='Code'] <- 'Codelist Code'
+  dat_term2 <- left_join(dat_term2,select(dat_term,'Codelist Code','CDISC Submission Value'),
+                         by='Codelist Code')
+  ctrl_variable <- domain_spec[domain_spec$'Controlled Terms, Codelist or Format' %in% term,]
+  ctrl_variable$'Controlled Terms, Codelist or Format' <- gsub("[()]","",
+                                                                 ctrl_variable$'Controlled Terms, Codelist or Format')
+  names(dat_term2)[names(dat_term2)=='CDISC Submission Value.y'] <- 'Controlled Terms, Codelist or Format'
+  dat_term2 <- left_join(dat_term2,select(ctrl_variable,'Controlled Terms, Codelist or Format',
+                                          'Variable Name'),by='Controlled Terms, Codelist or Format')
+  ctrl_variable <- ctrl_variable[ctrl_variable$'Variable Name' %in% names(data),]
+  noncontrol_col <- list()
+  for(i in ctrl_variable$'Variable Name'){
+    submission_value <- dat_term2[dat_term2$'Variable Name' %in% i,]$'CDISC Submission Value.x'
+    if((any(data[[i]] %in% submission_value) == F)){
+      noncontrol_col[[i]] <- i
+    }
+  }
+  noncontrol_col <- unlist(noncontrol_col)
+  if(length(noncontrol_col)>0){
+  stop(
+    "The following columns has entries not consistent with controlled terminology: ",
+    paste(noncontrol_col, collapse=", ")
+  )
+  }
+}
+
+# validate_ctrl_terminology(dat,domain_spec = domain_specs$DM)
+
+#' Check to see if columns are consistent with ISO format, give an error if columns are not consistent.
+#' 
+#' @param data The dataset whose columns to check.
+#' @param domain_spec The domain specification ("CO", "DM", "SE", "SM", "SV") for checking columns.
+#' @return Give an error if columns are not consistent with ISO format.
+#' @export
+validate_iso <- function(data,domain_spec){
+  if(domain_spec=="CO"){
+    data("CO")
+    domain_spec <- force(CO)
+  } else if (domain_spec=="DM"){
+    data("DM")
+    domain_spec <- force(DM)
+  } else if (domain_spec=="SE"){
+    data("SE")
+    domain_spec <- force(SE)
+  } else if (domain_spec=="SM"){
+    data("SM")
+    domain_spec <- force(SM)
+  } else if (domain_spec=="SV"){
+    data("SV")
+    domain_spec <- force(SV)
+  }
+  data("d.countries")
+  d.countries <- force(d.countries)
```

Comment:
You don't have to load "terminology" into the namespace of the function because it should already be available from the package namespace.

Also for best practices, please name variables with underscores instead of periods (`d_countries`) to prevent confusion about S3 class naming.